sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -0,0 +1,367 @@
+  joint_matrix_fill(Grp, joint_matrix<T2, UseT, nRows, nCols, LayoutT, Grp> &,
+                    const Type &);
+
+  template <typename Grp, typename jmType, typename Type, use UseL,
+            size_t nRows, size_t nCols, matrix::layout LayoutT,
+            access::address_space Sp, access::decorated IsDecorated,
+            std::enable_if_t<
+                std::is_same<jmType, std::remove_const_t<Type>>::value ||
+                    (std::is_same<jmType, precision::tf32>::value &&
+                     std::is_same<std::remove_const_t<Type>, float>::value),
+                bool>>
+  friend inline __SYCL_ALWAYS_INLINE void
+  joint_matrix_load(Grp,
+                    joint_matrix<jmType, UseL, nRows, nCols, LayoutT, Grp> &,
+                    multi_ptr<Type, Sp, IsDecorated>, size_t);
+
+  template <typename Grp, typename jmType, typename Type, size_t nRows,
+            size_t nCols, access::address_space Sp,
+            access::decorated IsDecorated,
+            std::enable_if_t<
+                std::is_same<jmType, std::remove_const_t<Type>>::value, bool>>
+  friend inline __SYCL_ALWAYS_INLINE void joint_matrix_load(
+      Grp,
+      joint_matrix<jmType, use::accumulator, nRows, nCols,
+                   sycl::ext::oneapi::experimental::matrix::layout::dynamic,
+                   Grp> &,
+      multi_ptr<Type, Sp, IsDecorated>, size_t,
+      sycl::ext::oneapi::experimental::matrix::layout);
+
+  template <typename Grp, typename Type, size_t nRows, size_t nCols,
+            access::address_space Sp, access::decorated IsDecorated>
+  friend inline __SYCL_ALWAYS_INLINE void joint_matrix_store(
+      Grp,
+      joint_matrix<Type, use::accumulator, nRows, nCols,
+                   sycl::ext::oneapi::experimental::matrix::layout::dynamic,
+                   Grp> &,
+      multi_ptr<Type, Sp, IsDecorated>, size_t,
+      sycl::ext::oneapi::experimental::matrix::layout);
+
+  template <typename Grp, typename T_a, typename T_b, typename T_c,
+            std::size_t m, std::size_t k, std::size_t n, layout LayoutA,
+            layout LayoutB>
+  friend inline __SYCL_ALWAYS_INLINE
+      joint_matrix<T_c, use::accumulator, m, n,
+                   sycl::ext::oneapi::experimental::matrix::layout::dynamic,
+                   Grp>
+      joint_matrix_mad(
+          Grp, joint_matrix<T_a, use::a, m, k, LayoutA, Grp> &,
+          joint_matrix<T_b, use::b, k, n, LayoutB, Grp> &,
+          joint_matrix<T_c, use::accumulator, m, n,
+                       sycl::ext::oneapi::experimental::matrix::layout::dynamic,
+                       Grp> &);
+
+public:
+  joint_matrix() {
+#ifndef __SYCL_DEVICE_ONLY__
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_ERROR_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  inline __SYCL_ALWAYS_INLINE decltype(auto) get_wi_data() {
+#if defined(__SYCL_DEVICE_ONLY__)
+#if defined(__NVPTX__)
+    return wi_data(cuda_impl.wi_marray);
```

Comment:
can wi_data move to unified?

Reply:
Yes, I've just moved it. For the constructor issue, let me know if you want me to add the constructor overload taking the Group sg argument, or whether you want to add it later and only have the constructor that doesn't take an argument for now.

Reply:
I think we should keep both in the implementation. 
@yubingex007-a11y, please add the constructor without Group argument in your unified PR as well to match this one.

Reply:
OK sure, I've added the constructor that takes a Group argument here.